Lib/inspect.py

```diff
@@ -2109,7 +2109,7 @@ def _signature_strip_non_python_syntax(signature):
     self_parameter = None
     last_positional_only = None
 
-    lines = [l.encode('ascii') for l in signature.split('\n')]
+    lines = [l.encode('ascii') for l in signature.split('\n') if l]
```

Comment:
`tokenize` wants something readline like, which is most idiomatically done by `.splitlines(keepends=True)`.

However, using that causes this test https://github.com/python/cpython/blob/16ae02e3be811e1c54fd253473774066520bcb70/Lib/test/test_inspect.py#L4127 to preserve newlines. I chose not to do this to make the diff smaller, but let me know if this is preferable. Doesn't result in any user visible difference.

Reply:
I think it's better to make it here so we don't need to change the test, even if the test is just testing a private function.